docs/pre-release/argo/canary-rollouts-with-argo.md

```diff
@@ -0,0 +1,262 @@
+
+## 1. Install and configure Edge Stack
+
+You'll need to first install Edge Stack in your cluster. Follow the [Edge Stack installation](/docs/latest/tutorials/getting-started) to install Edge Stack via Kubernetes YAML, Helm, or the command-line installer in your cluster.
+
+By default, Edge Stack routes via Kubernetes services. For best performance with canaries, we recommend you use endpoint routing. Enable endpoint routing on your cluster by saving the following configuration in a file called `resolver.yaml`:
+
+```
+apiVersion: getambassador.io/v2
+kind: KubernetesEndpointResolver
+metadata:
+  name: endpoint
+```
+
+Apply this configuration to your cluster: `kubectl apply -f resolver.yaml`.
+
+## 2. Install and configure Argo Rollouts
+
+If you're using Google Kubernetes Engine, grant your account the ability to create new Cluster Roles:
+
+```
+kubectl create clusterrolebinding YOURNAME-cluster-admin-binding --clusterrole=cluster-admin --user=YOUREMAIL@gmail.com
+```
+
+Now, install the Argo Rollouts controller in your cluster. Note that Argo must be installed in the `argo-rollouts` namespace.
+
+```
+kubectl create namespace argo-rollouts
+kubectl apply -n argo-rollouts -f https://raw.githubusercontent.com/datawire/argo-rollouts/ambassador/deploy/manifests/install.yaml
+```
+
+Finally, install the `kubectl` plugin, which will let you manage and visualize rollouts from the command line.
+
+Mac OS X:
+
+```
+brew install argoproj/tap/kubectl-argo-rollouts
+```
+
+Linux:
+
+```
+sudo curl -fL https://github.com/argoproj/argo-rollouts/releases/latest/download/kubectl-argo-rollouts-linux-amd64 -o /usr/local/bin/kubectl-argo-rollouts
+sudo chmod a+x /usr/local/bin/kubectl-argo-rollouts
+```
+
+Verify correct installation:
+
+```
+kubectl argo rollouts version
+```
+
+## 3. Create the Kubernetes Services
+
+We'll create two Kubernetes services, named `echo-stable` and `echo-canary`:
+
+```
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: echo
+  name: echo-stable
+spec:
+  type: ClusterIP
+  ports:
+  - name: http
+    port: 80
+    protocol: TCP
+    targetPort: 8080
+  selector:
+    app: echo
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: echo
+  name: echo-canary
+spec:
+  type: ClusterIP
+  ports:
+  - name: http
+    port: 80
+    protocol: TCP
+    targetPort: 8080
+  selector:
+    app: echo 
+```
+
+We'll also create an Edge Stack route to the services:
+
+```
+apiVersion: getambassador.io/v2
+kind:  Mapping
+metadata:
+  name:  rollouts-demo
+spec:
+  prefix: /demo
+  rewrite: /demo
+  service: echo-stable:80
+  resolver: endpoint
+```
+
+## 4. Deploy the Echo Service
+
+Argo Rollouts is orchestrated via the `Rollouts` resource. A `Rollout` resource is a superset of the standard Kubernetes `deployment` resource, and adds the ability to manage the process by which your service is deployed.
```

Comment:
As there is no direct relationship between a Rollout and a Deployment I think it is better saying something like:

_A `Rollout` resource is an alternative of the standard Kubernetes `Deployment`, adding the ability to run Canary or BlueGreen deployment strategies._